packages/hub/src/app/models/[owner]/[slug]/EditSquiggleSnippetModel.tsx

```diff
@@ -36,12 +39,66 @@ import {
   draftUtils,
   useDraftLocator,
 } from "./SquiggleSnippetDraftDialog";
+import { FormModal } from "@/components/ui/FormModal";
 
 export type SquiggleSnippetFormShape = {
   code: string;
   relativeValuesExports: RelativeValuesExportInput[];
 };
 
+type OnSubmit = (extraData?: { comment: string }) => Promise<void>;
+
+const SaveDialog: FC<{ onSubmit: OnSubmit; close: () => void }> = ({
+  onSubmit,
+  close,
+}) => {
+  type SaveFormShape = {
+    comment: string;
+  };
+  const form = useForm<SaveFormShape>(); // TODO
+
+  const handleSubmit = form.handleSubmit(async ({ comment }) => {
+    await onSubmit({ comment });
+    close();
+  });
+
+  return (
+    <FormModal
+      onSubmit={handleSubmit}
+      title="Save with comment"
+      form={form}
+      close={close}
+      submitText="Save"
+    >
+      <TextAreaFormField<SaveFormShape> name="comment" label="Comment" />
+    </FormModal>
+  );
+};
+
+const SaveButton: FC<{ onSubmit: OnSubmit; disabled: boolean }> = ({
+  onSubmit,
+  disabled,
+}) => {
+  return (
+    <ButtonWithDropdown
+      theme="primary"
+      size="small"
+      onClick={onSubmit}
+      disabled={disabled}
+      renderDropdown={({ close }) => (
+        <DropdownMenu>
+          <DropdownMenuModalActionItem
+            render={() => <SaveDialog onSubmit={onSubmit} close={close} />}
+            title="Save with comment..."
```

Comment:
Should I add an icon? I forgot we had `ChatBubbleLeftIcon` for docstrings which might be appropriate here.

Reply:
Yea, that seems good.